sdk/src/sanitize.rs

```diff
@@ -0,0 +1,21 @@
+#[derive(PartialEq, Debug)]
+pub enum SanitizeError {
+    Failed,
+    IndexOutOfBounds,
+    ValueOutOfRange,
+}
+
+pub trait Sanitize {
+    fn sanitize(&self) -> Result<(), SanitizeError> {
+        Ok(())
```

Comment:
I think providing a blanket implementation for `Sanitize` is best avoided. Instead, we should force to duplicate these nop sanitization if absolutely needed like for [`Pubkey`](https://github.com/solana-labs/solana/pull/9706/files#diff-b7e07a46126fae311692edc2141735ffR31).

It's pretty easy to forget to add an `impl` for `Sanitize` when adding a new `struct`. Because, engineers are lazy by nature they don't do anything when a compiler passes.. ;)

Reply:
You still have to write it into as empty.

Reply:
Why provide a default (and dummy) implementation of `sanitize`?

Reply:
Most thing just pass.  Like signature or pubkey

Reply:
I agree with @ryoqun I'd make it more explicit